hbw/tasks/ml.py

```diff
@@ -380,6 +399,22 @@ def run(self):
 
         # initialize the DatasetLoader
         ml_dataset = self.ml_model_inst.data_loader(self.ml_model_inst, process, events, stats)
+        # NOTE: Almost closure
+        sum_train_weights = np.sum(ml_dataset.train_weights)
+        n_events_per_fold = len(ml_dataset.train_weights)
+        logger.info(f"Sum of traing weights is: {sum_train_weights} for {n_events_per_fold} {process} events")
+
+        if self.ml_model_inst.config_inst.get_process(process).x("ml_config", None):
+
+            if self.ml_model_inst.config_inst.get_process(process).x.ml_config.weighting == "equal":
+                for sub_proc in self.ml_model_inst.config_inst.get_process(process).x.ml_config.sub_processes:
+                    proc_mask, sub_id = get_proc_mask(events, sub_proc, self.ml_model_inst.config_inst)
+                    xcheck_weight_sum = np.sum(ml_dataset.train_weights[proc_mask])
+                    xcheck_n_events = len(ml_dataset.train_weights[proc_mask])
+                    logger.info(
+                        f"For the equal weighting method the sum of weights for {sub_proc} is {xcheck_weight_sum} "
+                        f"(No. of events: {xcheck_n_events})",
+                    )
```

Comment:
this is just a cross-check, right? maybe add a note that this is clear
```suggestion
        # check that equal weighting works as intended
        if self.ml_model_inst.config_inst.get_process(process).x("ml_config", None):
            if self.ml_model_inst.config_inst.get_process(process).x.ml_config.weighting == "equal":
                for sub_proc in self.ml_model_inst.config_inst.get_process(process).x.ml_config.sub_processes:
                    proc_mask, sub_id = get_proc_mask(events, sub_proc, self.ml_model_inst.config_inst)
                    xcheck_weight_sum = np.sum(ml_dataset.train_weights[proc_mask])
                    xcheck_n_events = len(ml_dataset.train_weights[proc_mask])
                    logger.info(
                        f"For the equal weighting method the sum of weights for {sub_proc} is {xcheck_weight_sum} "
                        f"(No. of events: {xcheck_n_events})",
                    )
```